presto-hive/src/main/java/com/facebook/presto/hive/metastore/ThriftHiveMetastore.java

```diff
@@ -582,6 +591,16 @@ public void grantTablePrivileges(String databaseName, String tableName, String g
     public void revokeTablePrivileges(String databaseName, String tableName, String grantee, Set<PrivilegeGrantInfo> privilegeGrantInfoSet)
     {
         try {
+            Set<HivePrivilegeInfo> existingPrivileges = getTablePrivileges(grantee, databaseName, tableName);
+
+            Set<PrivilegeGrantInfo> privilegesToRevoke = privilegeGrantInfoSet.stream()
+                    .filter(requestedPrivilege -> existingPrivileges.containsAll(parsePrivilege(requestedPrivilege)))
```

Comment:
Regarding the renaming: yeah I agree that it is verbose, specially given that it is being used inside a Java streams API call. But I think just keeping it as `privilege` would be confusing. The verbosity here might actually help clarify that the variable here represents one of the privileges requested in the `revoke` statement. I will keep it as it is.

Regarding the filter condition: we want to revoke only those privileges which the user already possesses (if a user does not possess a privilege, there is no way to revoke it). So the condition here makes sure that only those privileges that already exist are passed to the `collect` call.